geo/src/algorithm/mod.rs

```diff
@@ -1,3 +1,6 @@
+/// Kernels to compute various predicates
+pub mod kernels;
```

Comment:
Is it useful for this to be public? 

If we keep it private we have more leverage to refactor the API later, it's fine for it to be public if there's a reason, just wanted to make sure.

Reply:
The main use of making Kernel public is for implementing the functionality for custom scalar types.  I think it should be fine to make it `pub(crate)` for now; and expose it once the traits mature.